### ISSUE TITLE
MA0048 - Improve filename readability for generic types

### ISSUE THREAD

**Issue by davidkallesen** [meziantou/Meziantou.Analyzer#213]
When having multiple generic classes of the same name, Visual Studio's "refactor to file"-feature will generate a files that not follow MA0048's compliant set.

Examples:
```csharp
namespace Testing
{
    // filename: Foo.cs 
    public class Foo// compliant
    {
    }

    // filename: Foo{T}.cs 
    public class Foo<T> // non compliant
    {
    }
    
    // filename: Foo{TParameters,TResult}.cs 
    public class Foo<TParameters, TResult> // non compliant
    {
    }
}
```

The existing solution hides the concrete implementation for readability.
Example:
* Foo`1.cs
* Foo`2.cs
makes it difficult to locate the correct file.

The rule should allow this as valid naming:
* Foo.cs 
* Foo{T}.cs 
* Foo{TParameters,TResult}.cs

**Comment by meziantou** [comment]
Thanks for your suggestion! I've opened a PR with the changes https://github.com/meziantou/Meziantou.Analyzer/pull/214.

**Comment by davidkallesen** [comment]
Thx. for fast implementation and it works 👍
